modules/aws/ec2/output.tf

```
@@ -1,9 +1,15 @@
 output "instance_public_ip_address" {
-  value       = aws_instance.ec2.public_ip
-  description = "This outputs the public IP associated with the EC2 instance. Note that this ouput will be the same as the elastic IP if `needs_elastic_ip` is set to `true`. This output is of type `string`."
+  value       = aws_eip.public_elastic_ip[0].public_ip
+  description = "This outputs the public IP associated with the EC2 instance. Note that this output will be the same as the elastic IP if `needs_elastic_ip` is set to `true`. This output is of type `string`."
 }
 
 output "instance_id" {
   value       = aws_instance.ec2.id
   description = "This outputs the unique ID of the EC2 instance."
+}
+
+output "private_key" {
+  value       = tls_private_key.private_key.private_key_pem
+  description = "This outputs the private key"
```

```suggestion
  description = "This outputs the private key."
```

I think descriptions need to be a sentence ie end with a full stop